src/main/io/rcdevice.c

```diff
@@ -165,23 +165,92 @@ static void runcamDeviceSendRequestAndWaitingResp(runcamDevice_t *device, uint8_
     responseCtx.parserFunc = parseFunc;
     responseCtx.device = device;
     responseCtx.protocolVer = RCDEVICE_PROTOCOL_VERSION_1_0;
-    memcpy(responseCtx.paramData, paramData, paramDataLen);
-    responseCtx.paramDataLen = paramDataLen;
+    if (paramData != NULL) {
+        memcpy(responseCtx.paramData, paramData, paramDataLen);
+        responseCtx.paramDataLen = paramDataLen;
+    }
     responseCtx.userInfo = userInfo;
     rcdeviceRespCtxQueuePushRespCtx(&watingResponseQueue, &responseCtx);
 
     // send packet
     runcamDeviceSendPacket(device, commandID, paramData, paramDataLen);
 }
 
+static void runcamDeviceParseV1DeviceInfo(rcdeviceResponseParseContext_t *ctx)
+{
+    if (ctx->result != RCDEVICE_RESP_SUCCESS) {
+        return;
+    }
+
+    runcamDevice_t *device = ctx->device;
+    device->info.protocolVer = RCDEVICE_PROTOCOL_RCSPLIT_VERSION;
+    device->info.features = RCDEVICE_PROTOCOL_FEATURE_SIMULATE_POWER_BUTTON | RCDEVICE_PROTOCOL_FEATURE_SIMULATE_WIFI_BUTTON | RCDEVICE_PROTOCOL_FEATURE_CHANGE_MODE;
+    device->isReady = true;
+}
+
+static uint8_t crc_high_first(uint8_t *ptr, uint8_t len)
+{
+    uint8_t i;
```

Comment:
Declare the variable where it's first used. Use `unsigned` for loop counters, as it yields more efficient code.